src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -319,9 +319,10 @@ ThreadSafeReference RealmCoordinator::get_unbound_realm()
 }
 
 void RealmCoordinator::do_get_realm(RealmConfig&& config, std::shared_ptr<Realm>& realm,
-                                    util::Optional<VersionID> version, util::CheckedUniqueLock& realm_lock)
+                                    util::Optional<VersionID> version, util::CheckedUniqueLock& realm_lock,
+                                    bool first_time_open)
 {
-    const auto db_opened_first_time = open_db();
+    open_db();
```

Comment:
I think we should still use this, otherwise sync open won't work. The way I see it, the new parameter should be optional, and if set, it takes precedence over db_opened_first_time.